### PR TITLE
LOC threshold for comments detector and implementation of switch statements detector

### PR DESCRIPTION
**Pull request by DosOvO**

### CODE REVIEW

File: libvcs4j-tools/src/main/java/de/unibremen/informatik/st/libvcs4j/spoon/codesmell/dispensable/CommentsDetector.java [hunk 1]
```diff
@@ -66,7 +76,7 @@ private void visitCtExecutable(final CtExecutable executable) {
                 .map(p -> p.getEndLine() - p.getLine() + 1)
                 .reduce(0, Integer::sum);
 
-        if (numberOfStatements == 0) {
+        if (numberOfStatements == 0 || numberOfStatements < locThreshold) {
```
Comment (msteinbeck) [inline libvcs4j-tools/src/main/java/de/unibremen/informatik/st/libvcs4j/spoon/codesmell/dispensable/CommentsDetector.java]: As far as I can see, `numberOfStatements == 0` is not necessary?

File: libvcs4j-tools/src/main/java/de/unibremen/informatik/st/libvcs4j/spoon/codesmell/ooabusers/SwitchStatementsDetector.java [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+    public <S> void visitCtSwitch(CtSwitch<S> switchStatement) {
+        super.visitCtSwitch(switchStatement);
+        switchStatement.getCases().forEach(caseStatement -> {
+            final int mccVal = mcc.MCCOf(caseStatement)
+                    .orElseThrow(IllegalStateException::new);
+            if (mccVal >= mccThreshold) {
+                addCodeSmell(switchStatement,
+                        Collections.singletonList(createMccMetric(mccVal)),
+                        createSignature(switchStatement).orElse(null));
+                return;
```
Comment (msteinbeck) [inline libvcs4j-tools/src/main/java/de/unibremen/informatik/st/libvcs4j/spoon/codesmell/ooabusers/SwitchStatementsDetector.java]: why `return`?

File: libvcs4j-tools/src/main/java/de/unibremen/informatik/st/libvcs4j/spoon/codesmell/ooabusers/SwitchStatementsDetector.java [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    }
+
+    @Override
+    public <S> void visitCtSwitch(CtSwitch<S> switchStatement) {
+        super.visitCtSwitch(switchStatement);
+        switchStatement.getCases().forEach(caseStatement -> {
+            final int mccVal = mcc.MCCOf(caseStatement)
+                    .orElseThrow(IllegalStateException::new);
+            if (mccVal >= mccThreshold) {
+                addCodeSmell(switchStatement,
```
Comment (msteinbeck) [inline libvcs4j-tools/src/main/java/de/unibremen/informatik/st/libvcs4j/spoon/codesmell/ooabusers/SwitchStatementsDetector.java]: This creates a code smell for each case in `switchStatement`. Instead, you have to check if all cases exceed the mcc threshold and, if true, create a single code smell.

**Comment by msteinbeck** [comment]
Is it ready to be merged?

**Comment by DosOvO** [comment]
Yes

**Pull request merged**